src/languages/ruby.js

```diff
@@ -11,13 +11,7 @@ export default function(hljs) {
   const regex = hljs.regex;
   const RUBY_METHOD_RE = '([a-zA-Z_]\\w*[!?=]?|[-+~]@|<<|>>|=~|===?|<=>|[<>]=?|\\*\\*|[-/+%^&*~`|]|\\[\\]=?)';
   // TODO: move concepts like CAMEL_CASE into `modes.js`
-  const CLASS_NAME_RE = regex.either(
-    /\b([A-Z]+[a-z0-9]+)+/,
-    // ends in caps
-    /\b([A-Z]+[a-z0-9]+)+[A-Z]+/,
-  )
-  ;
-  const CLASS_NAME_WITH_NAMESPACE_RE = regex.concat(CLASS_NAME_RE, /(::\w+)*/)
+  const CLASS_NAME_RE = /\b([A-Z]+[a-z0-9_]*)+[A-Z]*/;
```

Comment:
I think we need the `+`... On it's own, without other context, `A` is not a class, it's a constant.  Classes are CamelCase. Constants are UPCASE.

```suggestion
  const CLASS_NAME_RE = /\b([A-Z]+[a-z0-9_]+)+[A-Z]*/;
```

Reply:
Reverted to `+`